Issue #1 - Do more checks

reynoldtan opened the pull request:
- Amended the notice/warning area to include about file size and data
rows limit.

- Added a check to see if user has an ongoing posconvert job request.
This is done by checking the tripal job table for jobs created by user, by the module
and if the status is not completed, error or cancelled.

Please review the warning message top of the posconvert form.

To test Prevent user from running another job.
1. Posconvert a text data. When a tripal job has been registered, load the main posconvert form.
  It should display a message about ongoin posconvert job registered.
2. Once download is complete, reload the same page.
  It should load the form ready for the next posconvert.

To test 20000 line.
1. Please upload or if possible, paste text data into the text area.
   It should warn user about excessive rows.

Thanks!

reynoldtan committed:
Issue #1 - Do more checks

- Amended the notice/warning area to include about file size and data
rows limit.

- Added a check to see if user has an ongoing posconvert job request.

carolyncaron commented on posconvert.module:
```diff
@@ -236,7 +268,12 @@ function posconvert_form($form, &$form_state) {
       }
 
-      if ($cols < 2 || count($check_rows) <= 0) {
+      if ($lines > 19500) {
```
Why not use the posconvert_max_line_limit variable here?

reynoldtan replied:
I've created a function that will allow these values (line and size limit) to be accessible to the entire module where it is required.

carolyncaron commented on posconvert.module:
```diff
@@ -236,7 +268,12 @@ function posconvert_form($form, &$form_state) {
+      if ($lines > 19500) {
+        // Too many lines nearing the allowed line limit.
+        $form['ajax_wrapper']['error']['#markup'] =
+          '<div class="messages error">Text data is near the limit of allowed number of line. Please try again.</div>';
```
How about: "Your file exceeds the maximum number of lines. Please try a smaller file or contact us."

reynoldtan replied:
Replaced line to reflect this message.

carolyncaron commented on theme/posconvert_page.tpl.php:
```diff
@@ -11,7 +11,14 @@
+      <strong>This is a <em>Beta Release</em> and is still undergoing testing.</strong>
+      <em style="color: red; font-weight: bold">Important:</em>
+      Posconvert is limited to files that are less than <?php print $posconvert_max_file_limit; ?>b or <?php print number_format($posconvert_max_line_limit); ?> lines (not including the header).
+      It will take up to an hour to conver a file  of <?php print number_format($posconvert_max_line_limit); ?> lines. If you have a larger file, please
```
"conver" should be "convert" and we can probably drop "a file of" to shorten the message a bit.

carolyncaron commented on theme/posconvert_page.tpl.php:
```diff
@@ -35,6 +42,17 @@
+if ($posconvert_has_job) {
+?>
+
+  <div class="messages status warning">We have detected that you have an ongoing Position Convert request. Please allow thie request to complete before doing another conversion.</div>
```
"thie" should be "the"

reynoldtan replied:
Sorry for the typos. Rectified.

carolyncaron reviewed (changes requested):
With the requested changes, the code looks good to go! I have one question: Is it possible to check for the number of lines that **don't** begin with a hashtag (#)? Some VCF files can have unwieldy header lines, and these lines don't contribute to slowing the job down at all, so it would be ideal to exclude those lines from the line count.

reynoldtan committed:
Check spell complete

carolyncaron commented:
## Testing functionality
1. Preventing additional posconvert requests 
   - Works as expected - beautiful!
   - I would like to test to ensure that a different user wouldn't see the message if somebody else is already using posconvert, but the code already implies that it is user-specific. @laceysanderson  and I may have to coordinate to adequately test this but I don't think it should be a barrier to making the module live.
2. File size limit
   - I tried the following files near the line # limit: 20000, 19999, and 19500 lines. All three triggered the error message: `Text data is near the limit of allowed number of line. Please try again.` After looking at the code, the issue appears to be line 271. Please see my comment.
   - I tried a 92.6 Mb file, and I only get a "spinning circle" for 1 second and it disappears, and the upload dialogue remains.
   - If I try a 225 Mb instead, the spinner lasts for a few seconds more, and I get the following message: 
![screen shot 2017-06-07 at 3 12 26 pm](https://user-images.githubusercontent.com/7927760/26901648-caed5750-4b93-11e7-8af7-5c4d5941476d.png)
    The error message is correct, however the green checkmarks feel rather misleading in this case!
3. Other - I noticed that the file download box contained an empty "Summary" section. Perhaps some residual code ended up in this commit by accident? ;-)
![screen shot 2017-06-07 at 3 14 37 pm](https://user-images.githubusercontent.com/7927760/26901784-3f24af10-4b94-11e7-96b1-af25c65ea0ad.png)

reynoldtan committed:
Added a file size check

- Implemented hook_file_validate()
that will check the file size against the limit.

- Created a function that will return line limit and file size limit.
This will make these values available to other functions.

laceysanderson commented:
Works on the test data :-D 👍 Since this is quite old and the module is not used frequently, I'm calling that a good enough check. If it breaks anything, someone will hopefully poke us ;-)
